File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditAddAttendanceTimeEventDto.java
package de.helfenkannjeder.helfomat.api.organization.event;

import de.helfenkannjeder.helfomat.api.organization.AttendanceTimeDto;
import de.helfenkannjeder.helfomat.core.organization.OrganizationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditAddAttendanceTimeEventDto extends OrganizationEventDto {
    private int index;
    private AttendanceTimeDto attendanceTime;

    OrganizationEditAddAttendanceTimeEventDto() {
    }

    public OrganizationEditAddAttendanceTimeEventDto(OrganizationId organizationId, int index, AttendanceTimeDto attendanceTime) {
        super(organizationId);
        this.index = index;
        this.attendanceTime = attendanceTime;
    }

    public int getIndex() {
        return index;
    }

    public AttendanceTimeDto getAttendanceTime() {
        return attendanceTime;
    }

    @Override
    public <T> T visit(OrganizationEventDtoVisitor<T> visitor) {
        return visitor.visit(this);
    }

}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditAddAttendanceTimeEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.api.organization.AttendanceTimeDto
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditAddAttendanceTimeEventDto(
    override val organizationId: OrganizationId,
    val index: Int,
    val attendanceTime: AttendanceTimeDto,
    override val eventApplicable: EventApplicability
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}